utilities/ordered-set.metta

```diff
@@ -158,13 +161,16 @@
 ;; Returns: (OS Number) → Elements in set1 but not in set2
 ;; Example: (OS.difference (ConsOS 0 (ConsOS 1 NilOS)) (ConsOS 1 NilOS) NilOS) → (ConsOS 0 NilOS)
 (: OS.difference (-> (OS Number) (OS Number) (OS Number) (OS Number)))
-(= (OS.difference NilOS $set2 $result) $result)
-(= (OS.difference (ConsOS $head $tail) $set2 $result)
-   (if (OS.contains $set2 $head)
-       (OS.difference $tail $set2 $result)
-       (chain (OS.insert genericCompare $head $result) $newResult
-         (OS.difference $tail $set2 $newResult)
-       )
+(= (OS.difference $set1 $set2 $result)
+   (unify $set1 NilOS
```

Comment:
The same goes for this one as well. You can review your own PR before sending it over to make sure you can catch this kind of differences between your change and the one on the `ref/mettalog` branch.